client/fvESg_service.go

```diff
@@ -0,0 +1,389 @@
+	cont, err := sm.GetViaURL(dnUrl)
+	contList := models.ListFromContainer(cont,"fvRsCons")
+	st := &schema.Set{
+		F: schema.HashString,
+	}
+	for _, contItem := range contList{
+		dat := models.G(contItem, "tnVzBrCPName")
+		st.Add(dat)
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsScopeFromEndpointSecurityGroup( parentDn, tnFvCtxName string) error {
+	dn := fmt.Sprintf("%s/rsscope", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s","tnFvCtxName": "%s"
+								
+			}
+		}
+	}`, "fvRsScope", dn,tnFvCtxName))
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+	return nil
+}
+
+func (sm *ServiceManager) ReadRelationfvRsScopeFromEndpointSecurityGroup( parentDn string) (interface{},error) {	
+	dnUrl := fmt.Sprintf("%s/%s/%s.json",models.BaseurlStr,parentDn,"fvRsScope")
+	cont, err := sm.GetViaURL(dnUrl)
+	contList := models.ListFromContainer(cont,"fvRsScope")
+	if len(contList) > 0 {
+		dat := models.G(contList[0], "tnFvCtxName")
+		return dat, err
+	} else {
+		return nil,err
+	}
+}
+
+func (sm *ServiceManager) CreateRelationfvRsProtByFromEndpointSecurityGroup( parentDn, tnVzTabooName string) error {
+	dn := fmt.Sprintf("%s/rsprotBy-%s", parentDn, tnVzTabooName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
+			}
+		}
+	}`, "fvRsProtBy", dn))
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationfvRsProtByFromEndpointSecurityGroup(parentDn , tnVzTabooName string) error{
+	dn := fmt.Sprintf("%s/rsprotBy-%s", parentDn, tnVzTabooName)
+	return sm.DeleteByDn(dn , "fvRsProtBy")
+}
+
+func (sm *ServiceManager) ReadRelationfvRsProtByFromEndpointSecurityGroup( parentDn string) (interface{},error) {
+	dnUrl := fmt.Sprintf("%s/%s/%s.json",models.BaseurlStr,parentDn,"fvRsProtBy")
+	cont, err := sm.GetViaURL(dnUrl)
+	contList := models.ListFromContainer(cont,"fvRsProtBy")
+	st := &schema.Set{
+		F: schema.HashString,
+	}
+	for _, contItem := range contList{
+		dat := models.G(contItem, "tnVzTabooName")
+		st.Add(dat)
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsIntraEpgFromEndpointSecurityGroup( parentDn, tnVzBrCPName string) error {
+	dn := fmt.Sprintf("%s/rsintraEpg-%s", parentDn, tnVzBrCPName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
```

Comment:
We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.